lib/bpf/prog_load.h

```diff
@@ -0,0 +1,42 @@
+#ifndef LIB_BPF_PROG_LOAD_H_
+#define LIB_BPF_PROG_LOAD_H_
+
+#include <cstring>
+#include <string>
+#include <string_view>
+
+#include "lib/bpf/bpf.h"
+#include "lib/error/assign_or_return.h"
+#include "lib/error/status_or.h"
+#include "lib/posix/ifindex.h"
+#include "lib/posix/unique_file_descriptor.h"
+
+namespace bpf {
+
+// BPF_PROG_LOAD wrapper.
+//
+// TODO: Incomplete.
+//
+// See 'man 2 bpf' for details.
+inline error::StatusOr<posix::UniqueFileDescriptor> ProgLoad(
+    const uint32_t program_type, std::string_view instructions,
```

Comment:
Tighter types for program_type? And, program_flags?

Reply:
See the comment on 'int cmd' and let me know what you think

Reply:
This one is more grey, because ProgLoad() is no longer a 1:1 mapping to a syscall. What is the ProgLoad() called that I should use as an end user of the library that just wants to load a piece of eBPF code? What encapsulates (in the hide sense) this?

Reply:
The hope is that the user of a library is still a few levels above this. This is the part of the design that isn't clear to me yet. My approach for now is to wrap kernel functionality with names that mirror the syscall/command constants until it is easy and obvious what the final function should be called. At a high enough level we'll start to have more traditional OOP classes and users won't need to worry about things like file descriptors and syscalls.

I would like to consider the function names so far to be uncontroversial in the sense that they are mechanical transformations of names from kernel headers. If you have other ideas I'd love to hear them. I think namespaces are often a fantastic tool to solve problems like this.

Reply:
Yes, I think this is addressed by making it clear that the namespace is not for end user consumption (raw::).